core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -157,7 +158,7 @@ public void onNext(HttpObject o) {
                         state = State.NEEDS_DATA_OR_TRAILERS;
                     }
                     if (endOfStream) {
-                        setDone(false);
+                        setDone(true);
```

Comment:
How about also calling `HttpData.touch()` in `NEEDS_DATA_OR_TRAILERS` case and `AggregatedHttpResponseHandler`? If you think it's not relevant, I can create a PR that records in more places to track the leak.